### ISSUE TITLE
Bug: Patterns in arrays are not considered explicit match

### ISSUE THREAD

**Issue by mdjermanovic** [eslint/eslint#19814]
### Environment

Node version: v22.14.0
npm version: v10.9.2
Local ESLint version: v9.28.0
Global ESLint version: no
Operating System: windows

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```js
export default [
    {
        files: [["**/*.myjs", "src/**"]],
        rules: {
            "no-undef": 2
        }
    }
];
```
</details>

```js
// src/foo.myjs
foo
```

### What did you expect to happen?

I'd expect all .myjs files in `src` directory to be included in linting.

### What actually happened?

```
$ npx eslint src/foo.myjs

C:\projects\tmp\tmp\src\foo.myjs
  0:0  warning  File ignored because no matching configuration was supplied

✖ 1 problem (0 errors, 1 warning)
```

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/stackblitz-starters-vajknonf

### Participation

- [x] I am willing to submit a pull request for this issue.

### Additional comments

This is because the code that checks whether a pattern is an explicit match doesn't account for arrays in `files`:

https://github.com/eslint/rewrite/blob/cb858ffb8b77ea76187a857546c7a838a1fb4881/packages/config-array/src/config-array.js#L1013-L1054

If the order is changed to `files: [["src/**", "**/*.myjs"]]`, then it is considered explicit match, by chance due to the way arrays are coerced to strings.

**Comment by lumirlumir** [comment]
Nice catch, thanks 👍 I can reproduce it.

Since the behavior of `files: [["src/**", "**/*.myjs"]]` and `files: [["**/*.myjs", "src/**"]]` should be the same, it's indeed a bug. So, I’m marking it as accepted.

![Image](https://github.com/user-attachments/assets/208c461d-b3d9-4e0f-8e54-48a5753bd989)

![Image](https://github.com/user-attachments/assets/735bae17-e522-47d6-852d-4f5d7c6536ef)
